tfjs-core/src/util_base.ts

```diff
@@ -303,7 +303,7 @@ export function rightPad(a: string, size: number): string {
 
 export function repeatedTry(
     checkFn: () => boolean, delayFn = (counter: number) => 0,
-    maxCounter?: number): Promise<void> {
+    maxCounter?: number, scheduleFn?: Function): Promise<void> {
```

Comment:
environment.ts imports util_base, so we couldn't import environment in util_base, which will result in circular import. Thus I introduce scheduleFn here. If we need a custom scheduling function other than setTimeout, we can pass the function in. In our case, it's setTimeoutCustom() from WebGL side. In this way, repeatedTry() keeps to be platform independent.